.github/workflows/build.yml

```diff
@@ -28,23 +29,31 @@ jobs:
       with:
         repository: ruby/ruby
         path: ruby
+        fetch-depth: 0
     - name: Set latest_commit
       id: latest_commit
       working-directory: ruby
       run: echo "commit=$(git rev-parse HEAD)" >> $GITHUB_OUTPUT
-
+    - name: Set latest commit (3.4-asan)
```

Comment:
I wonder if it should be named `3.4-asan` or `asan-3.4`.
On one hand for https://github.com/ruby/setup-ruby?tab=readme-ov-file#supported-version-syntax if we consider asan an "engine" it should be `asan-3.4`.
OTOH it's a special case anyway and I think we should hardcode it like other special head builds (e.g. no version matching).
IOW its engine would be `asan-3.4` or `3.4-asan`, no "version" from setup-ruby POV.

It's also "3.4 + ASAN" so 3.4-asan is kinda intuitive.
Let's stick with that for now.